File: model/src/main/java/org/bandrsoftwares/celestialdiary/model/dto/establishment/EstablishmentPrestationDTO.java
package org.bandrsoftwares.celestialdiary.model.dto.establishment;

import org.bandrsoftwares.celestialdiary.model.dto.saleable.prestation.WrappedPrestationDTO;
import org.bandrsoftwares.celestialdiary.model.mongodb.establishment.EstablishmentPrestation;

public record EstablishmentPrestationDTO(double customPrice, boolean usingCustomPrice, WrappedPrestationDTO prestation) {

    public EstablishmentPrestationDTO(EstablishmentPrestation establishmentPrestation) {
        this(establishmentPrestation.getCustomPrice(), establishmentPrestation.getUsingCustomPrice(),
             new WrappedPrestationDTO(establishmentPrestation.getPrestation()));
    }
}

File: ui-frontend/src/data/model/establishment/establishment-prestation-dto.ts
import {EstablishmentSaleableDTO} from "./establishment-saleable-dto";
import {WrappedPrestationDTO} from "../saleable/prestation/wrapped-prestation-dto";

export interface EstablishmentPrestationDTO extends EstablishmentSaleableDTO {
  prestation: WrappedPrestationDTO
}
